src/hex/shape.rs

```diff
@@ -144,93 +140,90 @@ impl<T: Clone> HexShape<T> {
         };
 
         // Create our array.
-        let mut arr = Array::from_shape_simple_fn(size, &Tile::default);
+        let mut arr = Array::from_shape_simple_fn(size, || None);
 
         // Construct tiles that the shape contains.
         for coord in hexes {
-            arr[[coord.q as usize, coord.r as usize]] = constructor();
+            arr[[coord.q as usize, coord.r as usize]] = Some(constructor());
         }
 
         HexShape::new(Some(arr), Some(transform))
     }
 
     /// Create a line shape.
     ///
-    /// Given a size and direction, see [`ShapeArgs`], this will create a line.
+    /// Given a size and direction this will create a line.
     ///
     /// see [`Self::make_shape`] for more.
     ///
     /// ```
     /// use gridava::core::tile::Tile;
-    /// use gridava::hex::shape::{HexShape, ShapeArgs, shapeargs};
+    /// use gridava::hex::shape::HexShape;
     ///
     /// /// Creates a line of size 1, 0-1 inclusive, and sets the tiles to Some(1)
-    /// let my_shape = HexShape::make_line(shapeargs!(1, 0, true), || Tile::new(Some(1)));
+    /// let my_shape = HexShape::make_line(1, 0, true, || Tile::new(Some(1)));
     /// ```
-    pub fn make_line<F>(args: ShapeArgs, constructor: F) -> HexShape<T>
```

Comment:
In principle this feels like a reasonable change in interface, but I'm curious as to what the goal is more generally.

Reply:
In terms of function input? Yea, the annoying part is that adding the extra struct adds a typename into the namespace, and an extra variable to track. Another option could very well be just to take in a tuple and destructure it. I think doing that though loses the context the argument names provide.

Reply:
I prefer this option to the destructuring for the reason you give.